### PR TITLE
feat!: add `encode/decodeValueType` as public callable methods

### PR DESCRIPTION
**Pull request by CJ42**
### What kind of change does this PR introduce (bug fix, feature, docs update, ...)?

Feature

### What is the current behaviour (you can also link to an open issue here)?

The methods `encodeValueType` and `decodeValueType` are currently not available as public method part of the ERC725 class.

These methods can be useful to encode certain LSP2 value type raw. For example, a `uin128` (For LSP2 Array length) to obtain an exactly 16 bytes long value.

### What is the new behaviour (if this is a feature change)?

Make this method part of the class, so that they can be called and used.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -168,6 +169,21 @@ export class ERC725 {
 
     throw new Error(`Incorrect or unsupported provider ${providerOrRpcUrl}`);
   }
+
+  private getAddressAndProvider() {
```
Comment (CJ42) [inline src/index.ts:173]: @Hugoo I just moved this function at the top, to group the `private` methods together.

**Comment by frozeman** [comment]
we need to update the docs as well, when we do this change

File: src/index.ts (hunk 1 above)
Comment (Hugoo) [reply src/index.ts:173]: ok nice ty

**Comment by Hugoo** [comment]
Ready to merge - just waiting for conflict fix and docs update: https://github.com/ERC725Alliance/erc725.js/blob/develop/docs/classes/ERC725.md

File: docs/classes/ERC725.md [hunk 2]
```diff
@@ -766,86 +975,6 @@ myErc725.decodeMappingKey(
 
 ---
 
-## encodePermissions
```
Comment (CJ42) [inline docs/classes/ERC725.md:769]: @Hugoo this has not been deleted, just move at the top to sort function names correctly by alphabetical order.

**Pull request merged**